test/functional/counter-test.js

```diff
@@ -0,0 +1,63 @@
+import { assert } from 'chai';
+import sinon from 'sinon';
+import { Dispatcher } from '../../.dist/index';
+import CounterStore from './counter-store';
+
+describe('Counter', () => {
+
+    let spyA = sinon.spy();
+    let spyB = sinon.spy();
+
+    it('should start at 0', () => {
+        assert.strictEqual(CounterStore.getCount(), 0);
+    });
+
+    it('should add change listeners', () => {
+        CounterStore.addChangeListener(spyA);
+        CounterStore.addChangeListener(spyB);
+    });
+
+    it('should increment the default amount', () => {
+        Dispatcher.handleAction('INCREMENT');
+        assert.strictEqual(CounterStore.getCount(), 1);
+        assert.strictEqual(spyA.callCount, 1);
+        assert.strictEqual(spyB.callCount, 1);
+    });
+
+    it('should increment by a given amount', () => {
+        Dispatcher.handleAction('INCREMENT', {
+            amount: 5
+        });
+        assert.strictEqual(CounterStore.getCount(), 6);
+        assert.strictEqual(spyA.callCount, 2);
```

Comment:
A fresh spy is used for each `it` in the unit tests, but it seemed to make more sense for them to be cumulative in the functional tests.